lib/datasource/remote/api/http_repo/http_repository.dart

```diff
@@ -1,21 +1,40 @@
 import 'dart:async';
 
 import 'package:async/async.dart';
+import 'package:firebase_crashlytics/firebase_crashlytics.dart';
+import 'package:flutter/foundation.dart';
+import 'package:http/http.dart';
 import 'package:http/http.dart' as http;
 import 'package:seeds/datasource/remote/api/http_repo/seeds_scopes.dart';
 import 'package:seeds/datasource/remote/api/http_repo/seeds_tables.dart';
 import 'package:seeds/datasource/remote/firebase/firebase_remote_config.dart';
 import 'package:seeds/datasource/remote/util/response_extension.dart';
 
+class _HttpClient extends http.BaseClient {
```

Comment:
I am not sure we should fire every request into crashalytics

Reply:
Flutterfire docs says for log method:

> To avoid slowing down your app, Crashlytics limits logs to 64kB and deletes older log entries when a session's logs go over that limit.

Reply:
64 kB is quite a lot text I don't think be a problem also I did not notice any bottle neck in the app.

Reply:
Please let's not go overboard with Crashlytics. 

It's useful but we do not want to depend on it (what if crashlytics is down, do these still work?)

We also don't want to send all http requests through crashlytics. 

Only ones that have unexpected crashes. 

We also need to filter out expected crashes - example, network error because user's internet was down.